src/Workspaces/Core/Portable/Options/DocumentOptionSet.cs

```diff
@@ -2,7 +2,7 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#pragma warning disable RS0030 // Do not used banned APIs (backwards compatibility)
+#pragma warning disable RS0030 // Do not used banned APIs: DocumentOptionSet, Option<T>, PerLanguageOption<T>
```

Comment:
💭 I'm concerned about the scope of this directive. I'm thinking we should never apply it to more than one line at a time.

Reply:
Only doing so when the entire file implements backward compat code, like in this case.

Reply:
Applying on each line would make it unreadable